visidata/undo.py

```diff
@@ -37,14 +37,23 @@ def undo(vd, sheet):
     if not vd.options.undo:
         vd.fail("options.undo not enabled")
 
-    # don't allow undo of first command on a sheet, which is always the command that created the sheet.
-    for i, cmdlogrow in enumerate(sheet.cmdlog_sheet.rows[:0:-1]):
+    cmdlogrows = sheet.cmdlog_sheet.rows
```

Comment:
How about:

    cmdlogrows = itertools.filterfalse(lambda r: r.longname == 'set-option')
    for i, cmdlogrow in enumerate(reversed(cmdlogrows[1:])):

?

Reply:
Okay I've incorporated this change.
Right now `cmdlog.py` only adds `'set-option'` commands before the sheet creation command, but never after it. So `filterfalse()` is fine as long as that is true.
Is there any situation where user-created code might cause the insertion of `'set-option'` after sheet creation? If that ever happens, `row_idx` will not be correct and the undo log would be corrupted.
https://github.com/saulpw/visidata/blob/e4f603b7b73cb20710d49328dd89fa3eb066f844/visidata/undo.py#L49

Reply:
If an option gets set by the user, it should be added as a `set-option` command on the source sheet cmdlog.  (I think this doesn't happen currently, which would mean that in-session option changes aren't recorded properly for replay.)  So your original code might have been better, though I can't see it after the force push.  Sorry to make an incorrect suggestion if that's the case!

Also what's the difference between this `islice` and `[1:]`?

Reply:
Oops, I shouldn't have force pushed to overwrite the original code, then! That code skipped just the first sequential `set-option` commands. But it did it by calculating indices. It is shorter and cleaner to translate it to iterators as you suggested. So I've done that now with https://github.com/saulpw/visidata/pull/2244/commits/43ad4ae37d4652516d2d33739326308fa0b46ec0. 

As for the `islice`, it was needed because the generator from `filterfalse` didn't allow subscripts. But `next()` is all we actually need, so I got rid of the `islice`.